cherrymusicserver/migrations/__init__.py

```diff
@@ -28,4 +28,53 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>
 #
+""" Contains migrations of CherryMusic data; for database migrations, see
+    the database package.
 
+    A migration is a module named `migration_nnnn` , where `nnnn` is one of a
+    series of consecutive integers starting with `0001`.
+
+    A migration has a `do()` function which will be called to execute the
+    migration; the function is expected to decide whether the migration should
+    be carried out or not. If a migration requires manual intervention or needs
+    to be aborted for any reason, the function should print an actionable error
+    message and sys.exit with an error status.
+"""
+
+#python 2.6+ backward compability
+from __future__ import unicode_literals
+
+import os
+import pkgutil
+import re
+
+from backport import callable   # Py 3.0 and 3.1 compat
+
+
+_MIGRATIONS_PATH = os.path.dirname(__file__)
+_NAME_PATTERN = re.compile('^migration_\d{4}$')
+
+
+def check_and_migrate_all():
+    """ Runs all necessary migrations in alphabetical order.
+
+        The program can sys.exit if manual intervention is required.
+    """
+    for migration in load_migrations():
+        migration.do()
```

Comment:
I thought of migrations as instances of generic "Tasks", but that's probably not the best thought I ever had, so I'd say you're right! :)